spot_rl_experiments/spot_rl/skills/atomic_skills.py

```diff
@@ -406,130 +632,66 @@ def __init__(self, spot: Spot, config, use_policies=True):
 
         self.env = SpotPlaceEnv(config, spot)
 
-    def reset_env_and_policy(self, place_target, is_local):
+    def reset_env_and_policy(self, goal_dict: Dict[str, Any]) -> Any:
         """
         Resets the env and policy
 
         Args:
-            place_target (np.array([x,y,z])): Place target in either global frame or base frame of the robot
-            is_local (bool): Whether the place target is in the base frame of the robot
+            goal_dict: Dict[str, Any] containing necessary goal information for placing
+                                      Required key: "place_target" (x,y,z) where robot needs to place at
+                                                    "is_local" (bool) Whether the place target is in the local frame of the robot
 
         Returns:
             observations: Initial observations from the env
         """
+        place_target = goal_dict.get("place_target", None)
+        is_local = goal_dict.get("is_local", False)
+        if place_target is None:
+            raise KeyError(
+                "Error in Place.reset_env_and_policy(): place_target key not found in goal_dict"
+            )
+
         observations = self.env.reset(place_target, is_local)
         self.policy.reset()
 
         return observations
 
-    def execute_place(self, place_target_list, is_local=False):
+    def execute(self, goal_dict: Dict[str, Any]) -> Tuple[bool, str]:  # noqa
         """
-        Execute place for each place target in place_target_list
+        Place the object in hand list at the given place target
+
+        CAUTION: The robot will DROP the object from the place_target location, please use objects that are not fragile!
 
         Args:
-            place_target_list (list): List of place targets to go and place
-            is_local (bool): Whether the place target is in the local frame of the robot
+            goal_dict: Dict[str, Any] containing necessary goal information for placing
+                                      Required key: "place_target" (x,y,z) where robot needs to place at
+                                                    "is_local" (bool) Whether the place target is in the local frame of the robot
 
         Returns:
-            success_list (list): List of dicts containing the following keys:
-                - time_taken (float): Time taken to place the object
-                - success (bool): Whether the place was successful
-                - place_target (np.array([x,y,z])): Place target in base frame
-                - ee_pos (np.array([x,y,z])): End effector position in base frame
+            status (bool): Whether robot was able to succesfully execute the skill or not
+            message (str): Message indicating description of success / failure reason
```

Comment:
Could you also add this caution as a print/logger statement as a warning? That way user can see this while running the code if not while writing it up.

Reply:
Good idea.

Reply:
added